src/SFA.DAS.RoATPService.Application.Api/StartupConfiguration/Startup.cs

```diff
@@ -121,38 +132,37 @@ public IServiceProvider ConfigureServices(IServiceCollection services)
                 _logger.LogError(e, "Error during Startup Configure Services");
                 throw;
             }
-
-            return serviceProvider;
         }
 
-        private IServiceProvider ConfigureIOC(IServiceCollection services)
+        private void ConfigureDependencyInjection(IServiceCollection services)
         {
-            var container = new Container();
-
-            container.Configure(config =>
-            {
-                config.Scan(_ =>
-                {
-                    _.AssembliesFromApplicationBaseDirectory(c => c.FullName.StartsWith("SFA"));
-                    _.WithDefaultConventions();
-
-                    _.ConnectImplementationsToTypesClosing(typeof(IRequestHandler<>)); // Handlers with no response
-                    _.ConnectImplementationsToTypesClosing(typeof(IRequestHandler<,>)); // Handlers with a response
-                    _.ConnectImplementationsToTypesClosing(typeof(INotificationHandler<>));
-                });
-
-                config.For<IWebConfiguration>().Use(Configuration);
-                config.For<SingleInstanceFactory>().Use<SingleInstanceFactory>(ctx => t => ctx.GetInstance(t));
-                config.For<MultiInstanceFactory>().Use<MultiInstanceFactory>(ctx => t => ctx.GetAllInstances(t));
-                config.For<IMediator>().Use<Mediator>();
-                config.For<IDbConnection>().Use(c => new SqlConnection(Configuration.SqlConnectionString));
-
-                config.Populate(services);
-            });
-
-            return container.GetInstance<IServiceProvider>();
+            services.AddTransient<IHttpContextAccessor, HttpContextAccessor>();
+            services.AddTransient(x => Configuration);
+            services.AddTransient(c => new SqlConnection(Configuration.SqlConnectionString));
```

Comment:
Should we be doing this in the container if we're creating them in the repository classes?

Reply:
It took me ages to do it this way, I've no idea how to approach this suggestion! Let's discuss tomorrow AM.

Reply:
I think the SqlConnection registration may be redundant. It was registered in StructureMap to map to IDbConnection but not used anywhere, I don't think, as we're creating connections in the repos like this:
https://github.com/SkillsFundingAgency/das-roatp-service/blob/8b3f8e0d809aef2bb69b086fb049aaedf3b40b17/src/SFA.DAS.RoATPService.Data/LookupDataRepository.cs#L73

Reply:
Any you just mean the sqlConnection setup, right? Yeah, I think that can come out. I'll just run it and see it hangs together